tests/python/test_fake.py

```diff
@@ -10,7 +10,37 @@
 from torchdistx.fake import fake_mode, is_fake, meta_like
 
 
-def test_meta_like_works_as_expected():
+def test_fake_mode_returns_cuda_tensor_if_fake_cuda_is_true() -> None:
+    if torch.cuda.is_available():
+        pytest.skip("Can only be tested if CUDA is not available.")
+
+    with fake_mode(fake_cuda=True):
+        a = torch.ones([10], device="cuda")
+
+    assert a.device.type == "cuda"
+
+
+def test_fake_mode_raises_error_if_fake_cuda_is_false() -> None:
+    if torch.cuda.is_available():
+        pytest.skip("Can only be tested if CUDA is not available.")
+
+    with pytest.raises(AssertionError):
+        with fake_mode():
+            torch.ones([10], device="cuda")
+
+
+def test_cuda_tensor_raises_error_after_fake_mode() -> None:
+    if torch.cuda.is_available():
+        pytest.skip("Can only be tested if CUDA is not available.")
+
+    with fake_mode(fake_cuda=True):
```

Comment:
would we want to consider enabling `fake_cuda` by default or would this cause issues?

Reply:
I deliberately disabled it by default. I think such behavior should be explicitly set by the user; otherwise we might succeed constructing tensors even for cases where the user would expect them to fail. E.g. in normal circumstances if I am on a CPU-only machine, my expectation would be for a CUDA tensor initialization to fail. Fake mode by default should not influence such behavior.